json_serializable/lib/src/json_key_utils.dart

```diff
@@ -94,51 +95,84 @@ JsonKey _from(FieldElement element, JsonSerializable classAnnotation) {
     }
   }
 
-  final disallowNullValue = obj.getField('disallowNullValue').toBoolValue();
-  final includeIfNull = obj.getField('includeIfNull').toBoolValue();
-
-  if (disallowNullValue == true) {
-    if (includeIfNull == true) {
-      throwUnsupported(
-          element,
-          'Cannot set both `disallowNullvalue` and `includeIfNull` to `true`. '
-          'This leads to incompatible `toJson` and `fromJson` behavior.');
-    }
-  }
-
   return _populateJsonKey(
     classAnnotation,
     element,
     defaultValue: defaultValueLiteral,
-    disallowNullValue: disallowNullValue,
+    disallowNullValue: obj.getField('disallowNullValue').toBoolValue(),
+    encodeEmptyCollection: obj.getField('encodeEmptyCollection').toBoolValue(),
     ignore: obj.getField('ignore').toBoolValue(),
-    includeIfNull: includeIfNull,
+    includeIfNull: obj.getField('includeIfNull').toBoolValue(),
     name: obj.getField('name').toStringValue(),
     nullable: obj.getField('nullable').toBoolValue(),
     required: obj.getField('required').toBoolValue(),
   );
 }
 
+const _iterableOrMapChecker =
+    TypeChecker.any([coreIterableTypeChecker, coreMapTypeChecker]);
+
 JsonKey _populateJsonKey(
   JsonSerializable classAnnotation,
-  FieldElement fieldElement, {
+  FieldElement element, {
   Object defaultValue,
   bool disallowNullValue,
   bool ignore,
   bool includeIfNull,
   String name,
   bool nullable,
   bool required,
+  bool encodeEmptyCollection,
 }) {
+  if (disallowNullValue == true) {
```

Comment:
[nit] this method got pretty long - can/should anything get broken out?

Reply:
Will look into that in a follow-up

Reply:
actually, looking at it, I'm not sure it could be reasonably smaller. Maybe pull out the calculating of each field w/ associated logic? Hrm...